lib/actions/local/flash.ts

```diff
@@ -55,9 +55,9 @@ export const flash: CommandDefinition<
 
 		Examples:
 
-			$ balena local flash path/to/balenaos.img
-			$ balena local flash path/to/balenaos.img --drive /dev/disk2
-			$ balena local flash path/to/balenaos.img --drive /dev/disk2 --yes
+			$ balena local flash path/to/balenaos.{img|zip|gz|bz2}
```

Comment:
Sorry, that was a mistake, I meant `.xz` files, not `.lzma` files which have a different header.

Reply:
```
$ balena local flash path/to/balenaos.img{.zip|.gz|.bz2|.xz}
```
We should probably keep examples with `--yes` and `--drive`.
Note that `.zip` archives are different: they may contain multiple files, we'll flash the first file ending with `.img`, `.iso`, `.bin`, `.dsk`, `.hddimg`, `.raw`, `.dmg`, `.sdcard`, `.rpi-sdimg` or `.wic`.

Reply:
I guess technically speaking it's `path/to/balenaos.{img|zip|img.gz|img.bz2|img.xz}` ? But that's obviously a bit harder to read :wink:

Reply:
Is it "more technically correct" (while still easy to read) to replace the curly brackets with square brackets? -
```
$ balena local flash path/to/balenaos.img[.zip|.gz|.bz2|.xz]
```
Because in the (ba)sh ecosystem / man pages, square brackets denote optional values whereas curly brackets denote mandatory but alternative values (?) [citation needed].

Reply:
Thing is, the filenames will be one of:
```
balenaos.img
balenaos.zip
balenaos.img.gz
balenaos.img.bz2
balenaos.img.xz
```
so it's hard to come up with a "nice" pattern which accurately covers all filenames. Perhaps something like `path/to/balenaos.{zip|img[.gz|.bz2|.xz]}` but that's even uglier.

`.gz`, `.bz2` and `.xz` are only compression formats (i.e. only a single file can be compressed, so the file-format itself doesn't store any filename information), so compressing and uncompressing the file simply adds the suffix. Whereas `.zip` is an archive format (i.e. multiple files can be compressed, with the filenames stored inside the file-format) and so `balenaos.zip` "contains" `balenaos.img`.

Even if the balena CLI is _capable_ of flashing .bz2 and .xz etc. images, perhaps the documentation / usage message should only mention those file formats that balenaOS is actually downloadable in? :man_shrugging: 

I'll stop interfering now... :wink:

Reply:
@lurch, thanks. In my experience downloading images from the balena web dashboard (I use Chrome on MacOS), the file is saved to my Downloads folder with the extension '.img.zip', and when extracted it produces a single '.img' file. The notation `.img[.zip|.gz|.bz2|.xz]` includes `.img.zip` and `.img`, which covers my user experience. I'm not sure if the behaviour is the same for other browsers and platforms, but I think the square bracket notation I proposed is at least good enough to convey the idea that `balena local flash` accepts compressed and uncompressed files, so I'll merge it. :-)

Reply:
Ah, fair enough! I've never downloaded any balena images, so I didn't realise it offered downloads with a `.img.zip` extension. Apologies for any confusion I caused.